src/repos.cc

```diff
@@ -712,9 +717,22 @@ int repo_specs_to_aliases( Zypper & zypper, const std::vector<std::string> & rsp
  * \sa InitRepoSettings
  */
 
-template <class Container>
-void do_init_repos( Zypper & zypper, const Container & container )
+void do_init_repos( Zypper & zypper )
 {
+  // bsc#1222086: No need to try actions for non-root user which will not perform.
+  // Print repo metadata stats instead.
+  std::optional<Date> now;
+  std::optional<PropertyTable> mdstats;
+  if ( geteuid() != 0 ) {
+    MIL << "turn off refresh for not-root user" << endl;
+    zypper.configNoConst().no_refresh = true;
+    now = Date::now();
+    mdstats = PropertyTable();
+    mdstats->add( _("Repository"),
+                  // translator: used as a table header
+                  _("Time since last refresh") );
+  }
```

Comment:
Shouldn't we also check if a chroot is used instead of / ? A normal user might be able to refresh it

Reply:
Good point.